packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts

```diff
@@ -148,6 +148,29 @@ export class PostgresSqlBuilder extends WarehouseBaseSqlBuilder {
         return "'";
     }
 
+    escapeString(value: string): string {
+        if (typeof value !== 'string') {
+            return value;
+        }
+
+        return (
+            value
+                // Escape single quotes by doubling them (PostgreSQL standard)
+                .replaceAll("'", "''")
+                // PostgreSQL LIKE wildcards need to be escaped with backslashes
+                .replaceAll('\\', '\\\\') // Escape backslashes first
+                .replaceAll('%', '\\%') // Escape % wildcard
+                .replaceAll('_', '\\_') // Escape _ wildcard
+                // Remove SQL comments (-- and /* */)
+                .replace(/--.*$/gm, '')
+                .replace(/\/\*[\s\S]*?\*\//g, '')
+                // Remove null bytes
+                .replaceAll('\0', '')
+                // Trim whitespace after removing comments
```

Comment:
Trimming might cause problems whenwhite spaces are intentional. Searching for `Jo ` will result `Jose`.